AppKit/CPTrackingArea.j

```diff
@@ -0,0 +1,141 @@
+CPTrackingEnabledDuringMouseDrag    = 1 << 9;
+
+var CPTrackingAreaRectKey            = @"CPTrackinkAreaRectKey",
+    CPTrackingAreaOptionsKey         = @"CPTrackingAreaOptionsKey",
+    CPTrackingAreaOwnerKey           = @"CPTrackingAreaOwnerKey",
+    CPTrackingAreaUserInfoKey        = @"CPTrackingAreaUserInfoKey",
+    CPTrackingAreaReferencingViewKey = @"CPTrackingAreaReferencingViewKey";
+
+
+/*!
+ @ingroup appkit
+ 
+ A CPTrackingArea defines a region of view that generates mouse-tracking and
+ cursor-update events when the mouse is over that region.
+ */
+@implementation CPTrackingArea : CPObject
+{
+    CGRect                  _rect           @accessors(getter=rect);
+    CPTrackingAreaOptions   _options        @accessors(getter=options);
+    id                      _owner          @accessors(getter=owner);
+    CPDictionary            _userInfo       @accessors(getter=userInfo);
+    
+    CPView                  _referencingView;
+}
+
+
+#pragma mark -
+#pragma mark Initialization
+
+/*! Initializes and returns an object defining a region of a view to receive mouse-tracking events, mouse-moved events, cursor-update events, or possibly all these events.
+ */
+- (CPTrackingArea)initWithRect:(CGRect)aRect options:(CPTrackingAreaOptions)options owner:(id)owner userInfo:(CPDictionary)userInfo
+{
+    if (self = [super init])
+    {
+        if (!(options > 0))
+            [CPException raise:CPInternalInconsistencyException reason:"Invalid CPTrackingArea options"];
+
+        // Check options:
+        // - at least one in CPTrackingMouseEnteredAndExited, CPTrackingMouseMoved, CPTrackingCursorUpdate
+        // - exactly  one in CPTrackingActiveWhenFirstResponder, CPTrackingActiveInKeyWindow, CPTrackingActiveAlways
+        // - no check on CPTrackingAssumeInside, CPTrackingInVisibleRect, CPTrackingEnableDuringMouseDrag
+        
+        if (!((options & CPTrackingMouseEnteredAndExited) || (options & CPTrackingMouseMoved) || (options & CPTrackingCursorUpdate)))
+            [CPException raise:CPInternalInconsistencyException reason:"You must at least select a type of event message"];
+        
+        if ((((options & CPTrackingActiveWhenFirstResponder) > 0) + ((options & CPTrackingActiveInKeyWindow) > 0) + ((options & CPTrackingActiveAlways) > 0)) != 1)
+            [CPException raise:CPInternalInconsistencyException reason:"You must select one and only one scope of tracking"];
+
+        _rect     = aRect;
+        _options  = options;
+        _owner    = owner;
+        _userInfo = userInfo;
+        
+        _referencingView = nil;
+    }
+    
+    return self;
+}
+
+
+#pragma mark -
+#pragma mark Implementation
+
+- (void)_setReferencingView:(CPView)aView
+{
+    _referencingView = aView;
+}
+
+- (CPView)_referencingView
+{
+    return _referencingView;
+}
+
+- (BOOL)_isReferenced
+{
+    return (!!_referencingView);
+}
+
+#pragma mark -
+#pragma mark CPCoding
+
+- (id)initWithCoder:(CPCoder)aCoder
```

Comment:
These methods should be in a separate `CPCoding` category. And if you are going to implement this, you also need to implement the `copy` method in a separate `CPCopying` category.